### PR TITLE
Make graph metrics public

### PR DESCRIPTION
**Pull request by zslade**
### Type of PR

- [ ] BUG
- [x] FEAT
- [ ] MAINT
- [ ] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

Related PRs:
- Original metrics PR (cluster size and density): #1677
- Add node degree + cluster centralisation : #1806 
- Add 'is bridge': #1894
- Tweaks and fixes: #1763, #1887, #1982, #2015

### Give a brief description for the solution you have provided

This PR signals the end of the first development round of the graph metrics functionality and makes it public 🙌 

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Added tests (if appropriate)
- [x] Updated CHANGELOG.md (if appropriate)
- [x] Made changes based off the latest version of Splink
- [x] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [3.9.13] - 2024-03-04
 
 - Support for Databricks Runtime 13.x+
+- Added method for computing graph metrics ([#2027](https://github.com/moj-analytical-services/splink/pull/2027))
```
Comment (ADBond) [inline CHANGELOG.md:13]: This line should go above the `## [3.9.13] ...` section, under the `## Unreleased` heading

**Review by ADBond — APPROVED** [review]
Looks great, just one small change to the changelog.

Thanks for hunting down all the relevant PRs, very handy for reference!

**Comment by RobinL** [comment]
This is great!  Very exciting!

File: CHANGELOG.md (hunk 1 above)
Comment (zslade) [reply CHANGELOG.md:13]: My Bad! Not sure how I managed that...cheers Andy!

**Pull request merged**